### ISSUE TITLE
[Relay Theory Cognitive Formalization] Positive-Weight Robustness — does scalar robustness reduce to componentwise dominance?

### ISSUE THREAD

**Issue by rinsakamo** [rinsakamo/relay-lm#2737]
# Status / scope

Foundational **Relay Theory — Cognitive Theory lane** formalization/falsification child on protected `relay-theory`, following completed #2701 and #2734.

Fresh authority immediately before creation:

```text
relay-theory = 2284ab0bb74aed66cf4b91e4545a8d11d7caeb5c
tree         = 0d6377998b758dcbeb769a14a0f9db4dde780c13
#2701        = CLOSED / completed
#2734        = CLOSED / completed
#2153        = OPEN
#2168        = OPEN
```

Fresh searches for `scalarization Pareto weights dominance`, `all weights Pareto representation`, and `positive weights componentwise dominance` found #2209/#2145/#2172 and neighboring lineage, but no dedicated theorem owner for the exact bridge below.

Theory-only. No RelayLM `v1` / `v2` runtime/product mutation and no actual-model execution.

# Trigger

Two Level-B lines now meet:

```text
#2701
  multi-criterion efficacy can be incomparable;
  operational dominance is criterion/resource-frame relative;
  Pareto-style comparison supplies no universal scalar ranking.

#2734
  two fixed lossy representations can reverse strict ranking
  when only strictly positive future-workload weights are swapped.
```

The remaining Grand Null is:

> **If a candidate is claimed to be no worse under every strictly positive workload weighting, is that robustness a new higher-order property, or does it reduce exactly to ordinary componentwise loss dominance?**

A Null win is preferred.

# R0 — neutral two-coordinate loss profiles

Use only generic finite loss profiles:

```text
A = (a1, a2) : Nat × Nat
B = (b1, b2) : Nat × Nat
```

Reuse #2734's neutral weighted loss:

```text
L_w(A) = w1 * a1 + w2 * a2
```

with lower loss = better.

Define:

```text
PositiveWeightRobustNoWorse A B :=
  forall w1 w2 : Nat,
    0 < w1 ->
    0 < w2 ->
    L_(w1,w2)(A) <= L_(w1,w2)(B)
```

This is not probability, utility, intelligence, or a physical law. It is a finite scalarization property.

# R1 — componentwise dominance implies positive-weight robustness

Mechanically prove:

```text
a1 <= b1
and
a2 <= b2
----------------
PositiveWeightRobustNoWorse A B
```

This direction is ordinary monotonicity of multiplication/addition.

# R2 — positive-weight robustness implies first-coordinate dominance

Do not use zero weight to isolate a coordinate.

Assume, toward contradiction:

```text
b1 < a1
```

Choose strictly positive weights:

```text
w1 = b2 + 1
w2 = 1
```

Since `a1 >= b1 + 1`, the excess on coordinate 1 alone exceeds the entire possible `b2` contribution on the right, regardless of `a2 >= 0`.

Derive contradiction with robust no-worse.

Candidate result:

```text
POSITIVE_WEIGHT_ROBUSTNESS_IMPLIES_FIRST_COORDINATE_DOMINANCE
```

# R3 — symmetric second-coordinate dominance

Assume:

```text
b2 < a2
```

Choose:

```text
w1 = 1
w2 = b1 + 1
```

Derive contradiction.

Candidate result:

```text
POSITIVE_WEIGHT_ROBUSTNESS_IMPLIES_SECOND_COORDINATE_DOMINANCE
```

# R4 — exact equivalence

Combine the directions:

```text
PositiveWeightRobustNoWorse A B
iff
(a1 <= b1 and a2 <= b2)
```

Candidate scoped verdict:

```text
ALL_POSITIVE_WEIGHT_ROBUSTNESS_EQUALS_COMPONENTWISE_LOSS_DOMINANCE
```

This is the central theorem.

# R5 — relation to #2734

No contradiction exists.

#2734 demonstrates:

```text
incomparable error profiles
  (0,2) vs (2,0)
+ different positive weights
  -> ranking reversal
```

This Issue predicts:

```text
robust ranking across every positive weight
  -> incomparable profiles cannot qualify
  -> only componentwise loss dominance survives
```

Thus scalar-weight robustness does not rescue a total ordering where the underlying loss vectors trade off.

# R6 — relation to #2701

#2701 already established a criterion-indexed Pareto-like operational dominance relation.

This Issue should not duplicate its generic efficacy machinery. It adds only the scalarization bridge:

> **Being no worse under every strictly positive linear weighting of the two loss coordinates is equivalent to being no worse coordinatewise.**

Allowed synthesis:

```text
robust scalarization over all positive weights
  -> collapses to constituent vector dominance
```

not:

```text
Pareto dominance = Intelligence
```

# R7 — Intelligence boundary

This transaction cannot define Intelligence.

Allowed implication for #2153/#2168:

> **A proposed cross-frame `robust Intelligence score` does not earn an extra primitive merely by surviving all positive scalarizations of an already-declared constituent loss vector; in this finite scope that robustness is exactly the constituent vector dominance relation.**

Forbidden implications:

```text
Intelligence = Pareto dominance
all intelligence dimensions are known
all frame changes are linear scalarizations
human/LLM intelligence is two-dimensional
robustness across arbitrary WORLDs reduces to this theorem
```

A real higher-order macro could still earn itself by predicting/organizing trajectories not captured in the declared constituent vector.

# Anti-cheat rules

Invalid wins include:

```text
allow zero weights only
  -> call strictly-positive robustness proved

bound weights to a tiny favorable set
  -> call all-positive robustness proved

change A/B profiles with the weights
  -> robustness proved

add hidden scalar utility
  -> componentwise reduction proved

one Nat-pair theorem
  -> general convex multiobjective theorem claimed
```

# Formalization discipline

Prefer one tiny module importing #2734's `WorkloadWeightedCompression` helper.

No:

```text
sorry
admit
native_decide
new probability / measure machinery
substantive new axiom/dependency/choice/default
```

Use ordinary Nat order/arithmetic only. Do not pull in category or optimization libraries.

# First transaction acceptance

A mature first PR should mechanically establish at minimum:

1. neutral `PositiveWeightRobustNoWorse` over `Nat × Nat` loss profiles;
2. componentwise dominance -> robust no-worse;
3. robust no-worse -> first coordinate dominance using strictly positive `(b2+1,1)`;
4. robust no-worse -> second coordinate dominance using strictly positive `(1,b1+1)`;
5. exact iff with componentwise dominance;
6. optional #2734 `(0,2)` / `(2,0)` witness showing neither is robustly no-worse than the other, if cheap;
7. no cognitive ontology in theorem names;
8. no zero-weight shortcut in the main converse;
9. exact-head and merge-head regular CI + Lean build/kernel-check/axiom-audit GREEN.

# Claim hierarchy

```text
Level A
  positive-weight robust scalar-loss comparison

Level B
  equivalence with componentwise two-coordinate loss dominance

Level C
  robust scalar macro adds no extra ordering beyond declared constituent loss vector in this finite scope

Level D
  NONE

Level E
  NONE
```

Architecture consequence: **NONE**.

Refs #2701 #2704 #2734 #2735 #2153 #2168 #2141 #2209 #2145 #2142.

**Comment by rinsakamo** [comment]
## Terminal reconciliation — positive-weight robustness reduces to componentwise dominance

Fresh protected authority at reconciliation:

```text
relay-theory = 6e2f7259e3de7dde98330dfc8ff76549a362359f
tree         = 34f56ea060828673c6a80691aad5a5e0e4a1063c
PR #2741     = MERGED / squash
```

Exact PR head:

```text
1e9783359130db43af2bfe6e5aa2554b94b99cab
```

Exact-head verification:

- Relay Theory CI #261: pytest / minimum-supported / package-smoke / lint = 4/4 GREEN;
- Relay Theory Lean CI #227: build / independent kernel-check / axiom-audit = all GREEN.

Merge-head verification on `6e2f7259...`:

- Relay Theory CI #262: pytest / minimum-supported / package-smoke / lint = 4/4 GREEN;
- Relay Theory Lean CI #228: build / independent kernel-check / axiom-audit = all GREEN.

Merged diff is exactly two files:

```text
formal/relay_theory/RelayTheory/PositiveWeightRobustness.lean  +124
formal/relay_theory/RelayTheory.lean                            +1 import
```

No `sorry`, `admit`, `native_decide`, probability/measure machinery, optimization library, substantive new axiom/dependency, or architecture mutation was introduced.

### Level A — finite robustness relation

For two natural-number loss profiles `candidate, baseline : Nat × Nat`, the transaction defines:

```text
PositiveWeightRobustNoWorse candidate baseline :=
  forall w1 w2 : Nat,
    0 < w1 ->
    0 < w2 ->
    weightedWorkloadLoss (w1,w2) candidate <=
      weightedWorkloadLoss (w1,w2) baseline
```

So both coordinates must remain live under every tested scalarization.

### Level B — mechanically earned result

Lean proves both directions and therefore the exact equivalence:

```text
PositiveWeightRobustNoWorse candidate baseline
iff
(candidate.1 <= baseline.1 and candidate.2 <= baseline.2)
```

The converse does **not** isolate a coordinate with zero weight.

For first-coordinate dominance, a hypothetical violation

```text
baseline.1 < candidate.1
```

is separated using the strictly positive weights

```text
(baseline.2 + 1, 1)
```

and the symmetric second-coordinate argument uses

```text
(1, baseline.1 + 1)
```

Thus the anti-cheat requirement that every coordinate retain positive weight is mechanically respected.

Scoped Level-B verdict:

```text
ALL_POSITIVE_WEIGHT_ROBUSTNESS_EQUALS_COMPONENTWISE_LOSS_DOMINANCE
```

### Relation to #2734

#2734 established a finite tradeoff pair:

```text
(0,2) vs (2,0)
```

whose strict ranking reverses when positive workload weights are swapped.

#2737 supplies the complementary boundary:

```text
tradeoff / incomparable loss vectors
  -> cannot sustain one no-worse ranking for every positive weighting

no-worse under every strictly positive linear weighting
  -> exactly componentwise loss dominance
```

So requiring robustness across all positive scalarizations does not manufacture a new total ordering over underlying tradeoffs.

### Relation to #2701

#2701 already established criterion-indexed / Pareto-like operational dominance and the possibility of incomparability under multiple criteria.

This transaction adds only the finite scalarization bridge:

> **Within a two-coordinate Nat loss surface, all-positive linear-scalarization robustness contains no ordering beyond the constituent componentwise dominance relation.**

It does not identify Pareto dominance with Intelligence or efficacy ontology.

### Level C — permitted interpretation

Within this exact finite scope:

> **A higher-order scalar does not earn independent scientific content merely because it preserves one ordering under every strictly positive linear weighting of an already-declared constituent loss vector. That robustness is exactly the constituent vector dominance relation.**

For #2153/#2168/#2141 this increases the burden on an Intelligence-like macro: a robust predictive/emergent macro must organize or predict something not already exhausted by the declared constituent profile, or a stronger causal role must be independently earned.

### Explicit non-claims

This result is **not**:

- a probability or Bayesian theorem;
- a general rate-distortion theorem;
- a general convex/multiobjective representation theorem;
- evidence that all relevant cognitive dimensions are known;
- evidence that every WORLD/task/resource/value change is a linear scalarization;
- a theorem about arbitrary human or LLM intelligence;
- `Intelligence = Pareto dominance`;
- architecture authority.

Architecture consequence: **NONE**.

Refs #2741 #2734 #2735 #2701 #2704 #2153 #2168 #2141 #2209 #2145 #2142.
